pkg/infra/game/state/state.go

```diff
@@ -6,20 +6,56 @@ import (
 )
 
 type AgentState struct {
-	Hp           uint
-	Stamina      uint
-	Attack       uint
-	Defense      uint
-	BonusAttack  uint
-	BonusDefense uint
+	Hp          uint
+	Stamina     uint
+	Attack      uint
+	Defense     uint
+	WeaponInUse commons.ItemID
+	ShieldInUse commons.ItemID
+	Weapons     []commons.ItemID
+	Shields     []commons.ItemID
```

Comment:
These need to be [immutable](https://github.com/benbjohnson/immutable), since we now provide agents direct access to AgentStates which would be copied but since these are primitives (they would have direct access to modify these values).

Reply:
It also might make sense to make this some sort of sorted immutable collection but not super necessary (since I assume agents will care more about their best weaponry and their worst in the case of deception).